bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxBridgeHandler.java

```diff
@@ -0,0 +1,133 @@
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolaxBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class SolaxBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SolaxBridgeHandler.class);
+    private static final int INITIAL_SCHEDULE_DELAY_SECONDS = 5;
+
+    private SolaxConfiguration config;
+
+    private LocalHttpConnector localHttpConnector;
+
+    private @Nullable ScheduledFuture<?> schedule;
+
+    public SolaxBridgeHandler(Bridge bridge) {
+        super(bridge);
+
+        config = getConfigAs(SolaxConfiguration.class);
+        localHttpConnector = new LocalHttpConnector(config.password, config.hostname);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        int refreshInterval = config.refresh;
+        TimeUnit timeUnit = TimeUnit.SECONDS;
+        logger.debug("Scheduling regular interval retrieval on every {} {}", refreshInterval, timeUnit);
+        schedule = scheduler.scheduleWithFixedDelay(this::retrieveData, INITIAL_SCHEDULE_DELAY_SECONDS, refreshInterval,
+                timeUnit);
+    }
+
+    private void retrieveData() {
+        try {
+            String rawJsonData = localHttpConnector.retrieveData();
+            logger.debug("Raw data retrieved = {}", rawJsonData);
+
+            updateInverterData(rawJsonData);
+
+            Bridge bridge = getThing();
+            if (bridge.getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            logger.debug("Exception received while attempting to retrieve data via HTTP", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void updateInverterData(@Nullable String rawJsonData) {
+        LocalConnectRawDataBean inverterParsedData = LocalConnectRawDataBean.fromJson(rawJsonData);
+        Bridge bridge = getThing();
+        List<Thing> things = bridge.getThings();
+        if (inverterParsedData != null) {
+            for (Thing thing : things) {
+                ThingHandler handler = thing.getHandler();
+                if (handler instanceof InverterDataUpdateListener listener) {
+                    listener.updateListener(inverterParsedData);
+                }
+            }
+        } else {
+            logger.warn("Parsed bean from the raw JSON data is null. Rawdata={}", rawJsonData);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (schedule != null) {
```

Comment:
You most likely have a null annotation warning here, which can be resolved by:
```suggestion
        ScheduledFuture<?> schedule = this.schedule;
        if (schedule != null) {
```

Please also set `this.schedule = null` afterwards.